Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -364,68 +358,71 @@ public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot
                     await response.WriteAsync(text);
                 }
 
-                if (VerifySignature(request, response))
+                if (this.VerifySignature(request, response))
                 {
                     if (slackEvent.payload != null)
                     {
                         // handle interactive_message callbacks and block_actions
                         slackEvent = JsonConvert.ToString(slackEvent.payload);
-                        if (options.VerificationToken != null && slackEvent.token != options.VerificationToken)
+                        if (this.options.VerificationToken != null && slackEvent.token != this.options.VerificationToken)
                         {
                             response.StatusCode = 403;
                         }
                         else
                         {
                             Activity activity = new Activity()
                             {
-                                Timestamp = new DateTime(),
+                                Timestamp = default(DateTime),
                                 ChannelId = "slack",
                                 Conversation = new ConversationAccount()
                                 {
-                                    Id = slackEvent.Channel.Id
+                                    Id = slackEvent.Channel.Id,
                                 },
                                 From = new ChannelAccount()
                                 {
-                                    Id = (slackEvent.BotId != null) ? slackEvent.BotId : slackEvent.User.Id
+                                    Id = (slackEvent.BotId != null) ? slackEvent.BotId : slackEvent.User.Id,
                                 },
                                 Recipient = new ChannelAccount()
                                 {
-                                    Id = null
+                                    Id = null,
                                 },
                                 ChannelData = slackEvent,
-                                Type = ActivityTypes.Event
+                                Type = ActivityTypes.Event,
                             };
 
                             // Extra fields that do not belong to activity
-                            activity.Conversation.Properties["thread_ts"] = slackEvent.ThreadTs;
+                            activity.Conversation.Properties["thread_ts"] = ((dynamic)slackEvent)["event"].thread_ts;
```

Comment:
NOTE: We have to look for an alternative to the usage of _dynamic_ in this case

Reply:
why do we need to look for an alternative for this?